app/controllers/fhir/r4/api_controller.rb

```diff
@@ -66,6 +64,11 @@ def update
     # Try to update the resource
```

Comment:
The code section above this is not update but I think if the resource is not "patient", server should return either 404 (not found) or 422 (unprocessable entity). 401 (not authorized) could also be considered. 400 means the resource itself is bad.

400 Bad Request - resource could not be parsed or failed basic FHIR validation rules (or multiple matches were found for conditional criteria)
401 Not Authorized - authorization is required for the interaction that was attempted
404 Not Found - resource type not supported, or not a FHIR end-point
422 Unprocessable Entity - the proposed resource violated applicable FHIR profiles or server business rules